src/Mapper/Type/ObjectType.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace MyTarget\Mapper\Type;
+
+use Doctrine\Common\Annotations\Reader;
+use Doctrine\Instantiator\Instantiator;
+use Doctrine\Instantiator\InstantiatorInterface;
+use MyTarget\Mapper\Annotation\Field;
+use MyTarget\Mapper\Mapper;
+use MyTarget\Mapper\Exception\ClassNotFoundException;
+
+class ObjectType implements Type
+{
+    /**
+     * @var InstantiatorInterface
+     */
+    private $instantiator;
+
+    /**
+     * @var Reader
+     */
+    private $annotations;
+
+    public function __construct(Reader $annotations, InstantiatorInterface $instantiator = null)
+    {
+        $this->annotations = $annotations;
+        $this->instantiator = $instantiator ?: new Instantiator();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function hydrated($value, $type, Mapper $mapper)
+    {
+        if ( ! class_exists($type)) {
+            throw new ClassNotFoundException($type);
+        }
+
+        $class = new \ReflectionClass($type);
+        $result = $this->instantiator->instantiate($type);
+
+        do {
+            foreach ($class->getProperties() as $property) {
```

Comment:
That is a bug, too bad there is no way to call `getOwnProperties` like in js

I think this would fix it? `if ($property->class !== $class) continue;`